OutOfSchool/OutOfSchool.WebApi.Tests/Services/DraftStorage/DraftStorageServiceTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.BusinessLogic.Models.Workshops;
+using OutOfSchool.BusinessLogic.Services.DraftStorage;
+using OutOfSchool.BusinessLogic.Services.DraftStorage.Interfaces;
+using OutOfSchool.Redis;
+
+namespace OutOfSchool.WebApi.Tests.Services.DraftStorage;
+
+[TestFixture]
+public class DraftStorageServiceTests
+{
+    private Mock<IReadWriteCacheService> readWriteCacheServiceMock;
+    private Mock<ILogger<DraftStorageService<WorkshopBaseDto>>> loggerMock;
+    private IDraftStorageService<WorkshopBaseDto> draftStorageService;
+
+    [SetUp]
+    public void SetUp()
+    {
+        loggerMock = new Mock<ILogger<DraftStorageService<WorkshopBaseDto>>>();
+        readWriteCacheServiceMock = new Mock<IReadWriteCacheService>();
+        draftStorageService = new DraftStorageService<WorkshopBaseDto>(readWriteCacheServiceMock.Object, loggerMock.Object);
+    }
+
+    [Test]
+    public async Task RestoreAsync_WhenDraftExistsInCache_ShouldRestoreAppropriatedEntity()
+    {
+        // Arrange
+        var workshopDraft = new WorkshopBaseDto()
+        {
+            Title = "title",
+            Email = "myemail@gmail.com",
+            Phone = "+380670000000",
+        };
+        var expected = new Dictionary<string, string>()
+            {
+                {"ExpectedKey", "{\"Title\":\"title\",\"Email\":\"myemail@gmail.com\",\"Phone\":\"+380670000000\"}"},
+            };
+
+        readWriteCacheServiceMock.Setup(c => c.ReadAsync(It.IsAny<string>()))
+            .Returns(() => Task.FromResult(expected["ExpectedKey"]));
+
+        // Act
+        var result = await draftStorageService.RestoreAsync("ExpectedKey");
+
+        // Assert
+        readWriteCacheServiceMock.Verify(
+            c => c.ReadAsync(It.IsAny<string>()),
+            Times.Once);
+        result.Should().BeOfType<WorkshopBaseDto>();
+        Assert.AreEqual(workshopDraft.Title, result.Title);
```

Comment:
(Code style) `Should()` and `Assert()` in one test. Consider usage of `Should....Which` or `Should...Where` or `BeEquivalentTo`